torch/distributed/device_mesh.py

```diff
@@ -443,48 +443,58 @@ def __getitem__(self, mesh_dim_names: Union[str, Tuple[str]]) -> "DeviceMesh":
             submesh = _mesh_resources.create_child_mesh(self, mesh_dim_names)
             return submesh
 
-        def get_group(
-            self, mesh_dim: Optional[Union[int, str]] = None
-        ) -> Union[ProcessGroup, List[ProcessGroup]]:
+        def get_group(self, mesh_dim: Optional[Union[int, str]] = None) -> ProcessGroup:
             """
-            Returns a list of ProcessGroups corresponding to the mesh dimensions, or
-            returns a single ProcessGroup if mesh_dim is specified or the given mesh has
+            Returns a single ProcessGroup if mesh_dim is specified or the given mesh has
             only one mesh dimension.
 
             Args:
                 mesh_dim (str/int, optional): it can be the name of the mesh dimension or the index
                 of the mesh dimension. Default is None.
 
             Returns:
-                A list of :class:`ProcessGroup` object when `mesh_dim` is not specified for
-                a DeviceMesh with more than 1 dimension; otherwise, returns a single
-                :class:`ProcessGroup` object.
+                A :class:`ProcessGroup` object.
             """
             if not hasattr(self, "_dim_group_infos"):
                 raise RuntimeError("DeviceMesh process groups not initialized!")
 
-            if self.mesh.ndim == 1:
-                return not_none(
-                    _find_pg_by_ranks_and_tag(*self._dim_group_infos[0][:2])
+            if self.mesh.ndim > 1 and mesh_dim is None:
+                raise RuntimeError(
```

Comment:
do we want to raise the error here or just issue a deprecation warning?
- how many users do we think are already using this api and would hit this error

cc @wanchaol

Reply:
I have reviewed all the internal use cases that I can find by fbgs `mesh.get_group(` (unless people named it some other stuff otherwise) and look like they are all either: 
1) already calling get_group with mesh_dim specified 
2) calling get_group on a 1D child mesh without mesh_dim specified

These two cases won't be affected.

Reply:
ok. it may be fine.  if you want to derisk further against a revert, you could do a warning in this PR and stack a PR on top that changes the warning to an error. but i'll stamp to unblock.

Reply:
> ok. it may be fine. if you want to derisk further against a revert, you could do a warning in this PR and stack a PR on top that changes the warning to an error. but i'll stamp to unblock.

I think it should be fine. I am imported this PR as a diff to let internal tests run on it. Either way, even though if we do a warning, if it is returning a list at this point then it would also result in an error.

Reply:
yeah I think either warning/error works, although this is a corner case, it's techniquelly BC breaking, we would probably need to put line in the release notes to explain the changes